src/Uno.UWP/Media/Capture/CameraCaptureUI.iOS.cs

```diff
@@ -75,34 +81,113 @@ private async Task<StorageFile> CaptureFile(CancellationToken ct, CameraCaptureU
-
-					var correctedImage = FixOrientation(image);
+					if (result.ValueForKey(new NSString("UIImagePickerControllerOriginalImage")) is UIImage image)
+					{
+						var correctedImage = FixOrientation(image);
 
-					(Stream data, string extension) GetImageStream()
+						(Stream data, string extension) GetImageStream()
+						{
+							switch (PhotoSettings.Format)
+							{
+								case CameraCaptureUIPhotoFormat.Jpeg:
+									return (image.AsJPEG().AsStream(), ".jpg");
+
+								case CameraCaptureUIPhotoFormat.Png:
+									return (image.AsPNG().AsStream(), ".png");
+
+								default:
+									throw new NotSupportedException($"{PhotoSettings.Format} is not supported");
+							}
+						};
+
+						var (data, extension) = GetImageStream();
+						return await CreateTempImage(data, extension);
+					}
+					else
 					{
-						switch (PhotoSettings.Format)
+						var assetUrl = result[UIImagePickerController.MediaURL] as NSUrl;
+						PHAsset phAsset = null;
+
+						if (this.Log().IsEnabled(LogLevel.Debug))
 						{
-							case CameraCaptureUIPhotoFormat.Jpeg:
-								return (image.AsJPEG().AsStream(), ".jpg");
+							this.Log().Debug($"Asset url {assetUrl}");
+						}
 
-							case CameraCaptureUIPhotoFormat.Png:
-								return (image.AsPNG().AsStream(), ".png");
+						if (assetUrl is not null)
+						{
+							if (OperatingSystem.IsIOSVersionAtLeast(11, 0))
+							{
+								if (!assetUrl.Scheme.Equals("assets-library", StringComparison.OrdinalIgnoreCase))
+								{
+									var doc = new UIDocument(assetUrl);
+									var fullPath = doc.FileUrl?.Path;
+
+									return await ConvertToMp4(fullPath);
+								}
+
+								phAsset = result.ValueForKey(UIImagePickerController.PHAsset) as PHAsset;
+							}
+						}
 
-							default:
-								throw new NotSupportedException($"{PhotoSettings.Format} is not supported");
+						if (phAsset == null)
+						{
+#if !__MACCATALYST__
+							assetUrl = result[UIImagePickerController.ReferenceUrl] as NSUrl;
+
+							if (this.Log().IsEnabled(LogLevel.Debug))
+							{
+								this.Log().Debug($"Asset url {assetUrl}");
+							}
+
+							if (assetUrl != null)
+							{
+								phAsset = PHAsset.FetchAssets(new NSUrl[] { assetUrl }, null)?.LastObject as PHAsset;
+							}
+#endif
 						}
-					};
 
-					var (data, extension) = GetImageStream();
-					return await CreateTempImage(data, extension);
-				}
-				else
-				{
-					return null;
+						if (phAsset is not null)
+						{
+							var originalFilename = PHAssetResource.GetAssetResources(phAsset).FirstOrDefault()?.OriginalFilename;
+							return await ConvertToMp4(originalFilename);
+						}
+
+						if (this.Log().IsEnabled(LogLevel.Warning))
+						{
+							this.Log().LogWarning($"Could not determine asset url");
+						}
+					}
 				}
+
+				return null;
+			}
+		}
+
+		private async Task<StorageFile> ConvertToMp4(string originalFilename)
+		{
```

Comment:
the callers seem to be able to pass null for `originalFilename`
should we add a safe exit here?
```suggestion
		private async Task<StorageFile> ConvertToMp4(string originalFilename)
		{
			if (originalFilename == null)
			{
				return null;
			}
```
or add null-guard around the callers?